docs/semanticdb/specification.md

```diff
@@ -810,6 +811,12 @@ signatures in supported languages.
 
 `access`. [Access](#access) modifier of the corresponding definition.
 
+`documentation`. The documentation string associated with a symbol.
```

Comment:
I'm wondering if it makes sense to specify the format of the string in a separate field 🤔 

```proto
message SymbolInformation {
  // ...
  string documentation = 20;
  DocumentationFormat documentationFormat = 21;
}

enum DocumentationFormat {
  MARKDOWN = 0;
  HTML = 1;
  PLAINTEXT = 2;
  JAVADOC = 3;
  SCALADOC = 4;
}
```
What do you think?

Reply:
I was thinking about something like this, but only denoting the language that it comes from. Ultimately decided against it for the following reasons:

1. Looking at the different JVM language doc types (Javadoc, Scaladoc, KDoc, Groovydoc etc), they all seem to follow the baseline format and features. Hypothetically, if javadoc was the baseline, one could simply use a javadoc formatter (with *X*doc features copied verbatim)
2. Giving the responsibility of what format it should be to the consumer of a SemanticDB file rather than having them support all possible kinds. While technically you could convert from any format to the rest, consumers would probably prefer not to have to support them all, but be able to decide themselves based on it being _at least_ Javadoc. For additional features present in e.g. Scaladoc, one can switch based on the `langauge` field.

Reply:
There's quite a big difference between Scaladoc and Javadoc. My motivation to add the format is that there can be different indexers for the same language that emit different formats. For example, you could implement a tool that takes SemanticDB as input and converts Javadoc into HTML/Markdown.

I would love to require this field to be markdown formatted but it's actually quite complicated to render docstrings correctly (subclasses can inherit from superclasses, Scaladoc allows interpolating variables that can be defined in subclasses, ...). A compiler plugin may not have all of the necessary information.


cc/ @tgodzik do you have any thoughts?

Reply:
Alternative design to avoid the awkward `documentation` and `documentationFormat` sibling fields
```
message SymbolInformation {
  // ...
  Documentation documentation = 20;
}

message Documentation {
  enum Format {
    HTML = 0;
    MARKDOWN = 1;
    JAVADOC = 2;
    SCALADOC = 3;
    KDOC = 4;
  }
  string message = 0;
  Format format = 1;
}
```

Reply:
I like the second option and I think it would be super useful to have DocumentationKind similar to `MarkupKind` that we have in LSP.

Reply:
Updated spec with @olafurpg's example. Let me know if the associated spec docs are as expected :slightly_smiling_face: